infra/discovery/src/listener/listener.js

```diff
@@ -230,9 +230,8 @@ logger.info(
 )
 
 setNetwork(config.network)
-try {
-  main()
-} catch (err) {
-  logger.error('Error occurred in listener main() process')
-  logger.error(err)
-}
+main().catch(err => {
+  logger.error('Error occurred in listener main() process:', err)
+  logger.error('Exiting')
+  process.exit(-1)
```

Comment:
Should probably not use a negative number here.  Can cause unexpected behavior on *nix systems.  Use a number `> 0` to signal an error.

Reply:
good point. done.